### DISCUSSION TITLE
How to add list IPs for "Allow List"

### DISCUSSION THREAD

**Discussion by macduyhai** [drakkan/sftpgo#1428] [Q&A]
Im trying to add a IP in a Allow List screen But It not work ! 
I see a notice at the top "Allowlist disabled in your configuration"
How can I enable this ? How to edit sftpgo.json when I used image pull?
Thank you for help!
<img width="1270" alt="image" src="https://github.com/drakkan/sftpgo/assets/25715092/1d08cc10-6f2d-4abe-96fc-89075f991fb3">

**Comment by levelonestl** [comment]
Assuming you've got your /var/lib/sftpgo  directory mounted somewhere locally you'd add:

`-e SFTPGO_COMMON__DEFENDER__SAFELIST_FILE=/var/lib/sftpgo/safelist.json`

You can edit the corresponding section in the sftpgo.json as well but using the environment variables is more portable with docker.

Once you've got that you can create the file you use the following format example:

```
{
    "addresses":[
        "1.1.1.1",
        "8.8.8.8"
    ],
    "networks":[
        "130.211.0.0/22"
    ]
}
```

Then run your image

**Reply by macduyhai** [reply]
Thank you so much !

**Reply by macduyhai** [reply]
@levelonestl If I want to allow uploading of certain types of files: txt, excel... What should I do?

**Reply by levelonestl** [reply]
**Per-directory pattern restrictions** under Users and/or Groups.  Although you might want to reconsider doing that simply because some sftp/ftp clients upload files with temporary names (that may not meet your file extension requirements) and then rename them upon completion of the upload.

**Reply by macduyhai** [reply]
Yes, user can do this, If I limit size file upload,It may be the more optimal solution ? 
 "httpd.max_upload_file_size " Is it limited to web use only? and it still creates a file with a small capacity, with the same name as the error uploaded file
How do I limit the file upload size via ssh and web?

**Reply by levelonestl** [reply]
No idea https://github.com/drakkan/sftpgo/tree/main/docs

**Reply by macduyhai** [reply]
Do you any suggestion for me If I used quota user ? https://sftpgo.stoplight.io/docs/sftpgo/51bf7871b80fa-update-folder-quota-usage-limits

**Reply by macduyhai** [reply]
maybe it doesn't work

![Uploading image.png…]()

**Reply by macduyhai** [reply]
![image](https://github.com/drakkan/sftpgo/assets/25715092/3f069a1b-39aa-468d-a159-8734a26ccfe1)
